app/waf/src/repositories/common.ts

```diff
@@ -0,0 +1,28 @@
+import {Entity} from '@loopback/repository';
+import {DataObject, Options, DefaultCrudRepository} from '@loopback/repository';
+import uuid = require('uuid');
+
+export class CommonRepository<
+  T extends Entity,
+  ID
+> extends DefaultCrudRepository<T, ID> {
+  create(data: DataObject<T>, options?: Options): Promise<T> {
+    if (!('id' in data)) {
+      data = Object.assign(data, {id: uuid()});
+    }
+
+    if (!('createdAt' in data)) {
+      data = Object.assign(data, {createdAt: new Date().toISOString()});
+    }
+
+    return super.create(data, options);
+  }
+
+  updateById(id: ID, data: DataObject<T>, options?: Options): Promise<void> {
```

Comment:
Also for `updateByid` function. Once it's called, the `updatedAt` should be changed accordingly. 
So I think `if (!('updatedAt' in data)) {` is unnecessary.